external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaCommitsSource.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.kafka010
+
+import com.codahale.metrics.{Counter, MetricRegistry}
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.metrics.source.Source
+
+class KafkaCommitsSource extends Source with Logging{
+
+  private val prefix = "structured-streaming-driver-kafka-consumer"
```

Comment:
Not sure from where this name comes but other sources has different naming patterns.

Reply:
Yeah I saw the other sources name, are there any restrictions on naming. I tried to be very specific about where the metrics from because there are consumers all over the place.

Reply:
I'm feeling `structured-streaming` is not needed (not important), whereas other parts sound good to classify. If Spark doesn't have been classifying where the metric comes from in metric name, `driver` might be another one to get rid of.